arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,308 @@
+                }
+                // Add the symbol to the result collection and ignore any error.
+                if let symbol = symbol {
+                    symbols.append((category, symbol))
+                }
+            }
+        }
+        getSymbolsGroup.notify(queue: .main) { [weak self] in
+            self?.symbols = symbols.sorted { $0.0 < $1.0 }
+            completion()
+        }
+    }
+    
+    /// Create an `AGSUniqueValueRenderer` to render feature layer with symbol styles.
+    /// - Parameter fieldNames: The attributes to match the unique values against.
+    /// - Returns: An `AGSUniqueValueRenderer` object.
+    func makeUniqueValueRenderer(fieldNames: [String]) -> AGSUniqueValueRenderer {
+        let renderer = AGSUniqueValueRenderer()
+        renderer.fieldNames = fieldNames
+        renderer.uniqueValues = symbols.flatMap { category, symbol in
+            // For each category value of a symbol, we need to create a
+            // unique value for it, so the field name matches to all categories.
+            category.symbolCategoryValues.map { symbolValue in
+                AGSUniqueValue(description: "", label: category.symbolName, symbol: symbol, values: [symbolValue])
+            }
+        }
+        return renderer
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["CreateSymbolStylesFromWebStylesViewController"]
+        
+        // Get the symbols from the symbol style hosted on an ArcGIS portal.
+        getSymbols(symbolStyle: symbolStyle, categories: SymbolCategory.allCases) { [weak self] in
+            guard let self = self else { return }
+            self.featureLayer.renderer = self.makeUniqueValueRenderer(fieldNames: ["cat2"])
+            self.legendBarButtonItem.isEnabled = true
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mapScaleObservation = mapView.observe(\.mapScale, options: .initial) { [weak self] (_, _) in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                self.featureLayer.scaleSymbols = self.mapView.mapScale >= 8e4
+            }
+        }
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        mapScaleObservation = nil
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "LegendTableSegue",
+           let controller = segue.destination as? UITableViewController {
+            controller.presentationController?.delegate = self
+            // The data source for the legend table display.
+            symbolsDataSource = SymbolsDataSource(tableView: controller.tableView, symbols: symbols)
```

Comment:
I realized that then the legend view controller is dismissed, the data source hangs around. You might want to add an unwind segue to set the `symbolsDataSource` to `nil` when the legend view controller goes away.

Reply:
I added an unwind segue action to release the data source. 
However it would cause the icon swatches to be downloaded each time we open the legend table. Is that OK?

Reply:
Ah, I didn't think of that. In that case, I suppose it makes sense to keep the data source around. Either that or retrieve them _before_ creating the data source, in which case the data for each row might look something like this:

```swift
struct LegendItem {
    var name: String
    var image: UIImage
}
```

Reply:
Sounds good. I'll let you know once it's ready for review!